fgpyo/alignment/__init__.py

```diff
@@ -0,0 +1,232 @@
+from typing import Callable
+from typing import List
+
+from fgpyo.sam import Cigar
+from fgpyo.sam import CigarElement
+from fgpyo.sam import CigarOp
+
+
+class Alignment:
+    """
+    A general class to describe the alignment between two sequences or partial ranges thereof
+    """
+
+    def __init__(
+        self,
+        query: str,
+        target: str,
+        query_start: int,
+        target_start: int,
+        cigar: Cigar,
+        score: int,
+    ):
+        """
+        Args:
+            query: the query sequence
+            target: the target sequence
+            query_start: the 1-based position in the query sequence where the alignment begins
+            target_start: the 1-based position in the target sequence where the alignment begins
+            cigar: a Cigar object describing the alignment of the two sequences
+            score: the alignment score
+        """
+        self.query: str = query
+        self.target: str = target
+        self.query_start: int = query_start
+        self.target_start: int = target_start
+        self.cigar: Cigar = cigar
+        self.score: int = score
+
+    @property
+    def query_end(self) -> int:
+        """
+        Returns:
+            One based closed coordinate of the end of the alignment on the query sequence.
+        """
+        return self.query_start + self.cigar.length_on_query() - 1
+
+    @property
+    def target_end(self) -> int:
+        """
+        Returns:
+            One based closed coordinate of the end of the alignment on the target sequence.
+        """
+        return self.target_start + self.cigar.length_on_target() - 1
+
+    def padded_string(
+        self,
+        match_char: str = "|",
+        mismatch_char: str = ".",
+        gap_char: str = " ",
+        pad_char: str = "-",
+    ) -> List[str]:
```

Comment:
prefer a tuple when we have a fixed # of items returned (or use a namedtuple/dataclass, but a tuple is best here).  That allows us to unpack on assignment: `q, a, t = alignment.padded_string(...)`
```suggestion
    ) -> Tuple[str, str, str]:
```